src/world.rs

```diff
@@ -219,12 +219,12 @@ impl World {
         }
         let mut gens = self.generations.write().unwrap();
         let mut gen = &mut gens[entity.get_id() as usize];
-        assert!(*gen > 0);
+        assert!(gen.is_alive());
         let mut app = self.appendix.write().unwrap();
         if entity.get_id() < app.next.get_id() {
-            app.next = Entity(entity.0, *gen+1);
+            app.next = Entity(entity.0, Generation(gen.0 + 1));
```

Comment:
This was rather a hack - to do `gen.0+1` before the `gen` is inverted. Instead, we should do:
```rust
gen = gen.die();
...
app.next = Entity(..., gen.raise());
```